### ISSUE TITLE
ErrorCode: "NullReferenceException",

### ISSUE THREAD

**Issue by ianTjor** [NCRCounterpointAPI/APIGuide#112]
We're getting this Response Status below. Our programming team has stated this looks more like an internal error since it's not providing system error tracking details. I can provide further detail on the call and details if needed. 

o	ErrorCode: "NullReferenceException",
o	Message: "An error occurred while trying to access an object of type: UserRoles, Object value is null, at CPAPI.Common.RepositoryCompanyBase.GetCacheKeyName(Object[] cacheKeyId)",
o	StackTrace: " at CPAPI.Common.RepositoryCompanyBase.GetCacheKeyName(Object[] cacheKeyId) at CPAPI.Common.KeyValueDataRepository.GetValueByName[T](String key) at CPAPI.Common.UserRoleBusiness.CheckPermissions(String userId, String path, String verb) at CPAPI.Common.RoleAttribute.Execute(IRequest req, IResponse res, Object requestDto) at ServiceStack.RequestFilterAttribute.RequestFilter(IRequest req, IResponse res, Object requestDto) at ServiceStack.ServiceStackHost.ApplyRequestFiltersSingle(IRequest req, IResponse res, Object requestDto) at ServiceStack.ServiceStackHost.ApplyRequestFilters(IRequest req, IResponse res, Object requestDto) at ServiceStack.Host.RestHandler.ProcessRequestAsync(IRequest httpReq, IResponse httpRes, String operationName)"

**Comment by Ho0m3s** [comment]
@ianTjor Did you ever find out what was causing this? Or how to fix it? I am having the same issue when using cURL to connect to our NCR API server. I have given the API user the right privileges in the API Admin console  for the endpoint i am trying to reach.

**Comment by ianTjor** [comment]
I've never recieved a comment back via Email or Github and the API is still non-functional.

**Comment by ccbradley** [comment]
Getting the exact same error code - but it happened "out of the blue". Did this ever get resolved?

**Comment by ccbradley** [comment]
> I've never recieved a comment back via Email or Github and the API is still non-functional.

yea - having same issue "out of the blue"..is the API still non-functional?

**Comment by ianTjor** [comment]
No, CounterPoint has never responded to any of my inquires. It seams all the support is managed by their vendors. I worked with two vendors. One was not equipped or up to speed with the API setup and processing. We got through development though. That said, the second vendor was great. We changed the setup to their best practices and suggestions. Including the processing procedure. All issues around the API were quickly clarified and addressed. Apparently there is some reporting and details available that are not present in error code returns. If we had started out with the second vendor we would have saved many billable hours. The qualified vendor is RELENTLESS.NET. Just thought I would give them some props. That said, the errors pointed to a few things and were resolved their involvement in the project. 

To this issue:
"out of the Blue" suggests to me, if still encountering the issue, that a good place to start is reviewing permissions settings in CounterPoint for access levels and permissions structure.

**Comment by mj185111** [comment]
Hello! 
A few things to check first:
-base64 encoding for the authorization at the header when sending requests.
-Do you have a role created with permissions set for that company and assigned to the user making the API call?

**Comment by golaod** [comment]
@mj185111 we went through the documentation steps and properly created API console admin, we uploaded api certificate, then we created a company and assigned admins to that company. We were using API for 1.5 month successfully and then it stopped working, so it's nothing on our side as the client was written once at the beginning and never changed.

**Comment by ccbradley** [comment]
@golaod is the lead on this .. we're on same team. @mj185111 your suggestions are not cause of issue. Any other suggestions?

**Comment by ianTjor** [comment]
To clarify, are you saying you confirmed the user rights within CounterPoint? Those can easily be changed by a user.

**Comment by ccbradley** [comment]
correct

**Comment by arpandalalplantt** [comment]
@ianTjor @Ho0m3s - How did you resolve this issue? I am having the same issue. I am using Postman to test the calls. I have added a role, assigned the endpoints and I am using a company login. I am running into the same stack-trace that @ianTjor has posted earlier. The endppoints /Items /Roles /Item/{ItemNo} fails for me. The endpoints that don't need company login (APIKeys) works.

**Comment by arpandalalplantt** [comment]
Looks like my problem above got resolved after I sent the ServerCache parameter with the request. I can not get response from endppoints /Items /Roles,  /Item/{ItemNo} and others.

**Comment by cyrusbaratt** [comment]
Could you please let us know how you pass the ServerCache parameter?
Thank you much
